exo/topology/ring_memory_and_flops_weighted_partitioning_strategy.py

```diff
@@ -0,0 +1,26 @@
+from typing import List
+from .partitioning_strategy import PartitioningStrategy
+from .topology import Topology
+from .partitioning_strategy import Partition
+
+
+class RingMemoryAndFlopsWeightedPartitioningStrategy(PartitioningStrategy):
+  def __init__(self, memory_weight: float = 0.5, flops_weight: float = 0.5):
+    self.memory_weight = memory_weight
+    self.flops_weight = flops_weight
+
+  def partition(self, topology: Topology) -> List[Partition]:
+    nodes = list(topology.all_nodes())
+    max_flops = max(node[1].flops for node in nodes)
+    max_memory = max(node[1].memory for node in nodes)
+    def compute_score(node):
+      return self.flops_weight * node[1].flops/max_flops + self.memory_weight * node[1].memory/max_memory
```

Comment:
many setups without GPUs (or without _recognized_ GPUs) default to 0, so be careful of divide by zero errors here